### PR TITLE
Migrate from html5lint to HTMLHint - fixes #100

### LINKED ISSUE #100: Replace html5-lint dependency
See #98. The server html5-lint was using has been shut down, so we need to find another way to do HTML validation for the tests.
I am very open to suggestions here!

### PR DESCRIPTION
**Pull request by agauniyal**
Tests pass, updated package.json and included .htmlhintrc

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +26,7 @@
   },
   "homepage": "https://zen-audio-player.github.io",
   "devDependencies": {
-    "html5-lint": "^0.2.5",
+    "htmlhint": "latest",
```
Comment (shakeelmohamed) [inline package.json]: Can we at least stick to the `0.9.x` line? I don't want something crazy to happen if they go to `1.0.0`

File: .htmlhintrc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "tag-pair": true,
+    "spec-char-escape": true,
+    "id-unique": true,
+    "src-not-empty": true,
+    "attr-no-duplication": true,
+    "title-require": true,
+    "head-script-disabled": false,
+    "alt-require": true,
+    "doctype-html5": true,
+    "id-class-value": 'dash',
```
Comment (shakeelmohamed) [inline .htmlhintrc]: nit: single quotes

File: .htmlhintrc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "style-disabled": true,
+    "inline-style-disabled": true,
+    "inline-script-disabled": true,
+    "space-tab-mixed-disabled": true,
+    "id-class-ad-disabled": true,
+    "href-abs-or-rel": 'abs',
```
Comment (shakeelmohamed) [inline .htmlhintrc]: nit: single quotes

File: package.json [hunk 4]
```diff
@@ -4,7 +4,9 @@
   "description": "Listen to YouTube videos, without the distracting visuals.",
   "main": "tests.js",
   "scripts": {
-    "pretest": "jshint js test",
+    "lint:js": "jshint js test",
+    "lint:html": "htmlhint index.html",
+    "pretest": "npm run lint:js && npm run lint:html",
```
Comment (shakeelmohamed) [inline package.json]: I like the way you broke this up into multiple scripts. For `pretest`, can we do something like `npm run test -- --grep="lint"`. It seems possible as of npm v2.0.0

File: test/html-check.js [hunk 5]
```diff
@@ -1,36 +0,0 @@
...
-                    var lineNumber = msg.lastLine;
-                    console.log("HTML5 Lint [%s]: %s %s (line %s)", type, message, lineNumber);
-                    assert.notStrictEqual(type, "error", message + ", see line " + lineNumber + " in index.html");
-                }
-            }
-            done();
-        });
-    });
-    it("should not contain tab characters", function() {
-        assert.strictEqual(-1, _html.indexOf("\t"), "Please remove any tab indents from index.html");
```
Comment (shakeelmohamed) [inline test/html-check.js]: I'm assuming `space-tab-mixed-disabled` will take care of this case

File: package.json (hunk 4 above)
Comment (agauniyal) [inline package.json]: Yup works that way although I wasn't aware of it :smile:

**Comment by agauniyal** [comment]
I've updated package.json file and had to turn off some checks which were interferring with jquery and absolute/relative url consistencies in this project.

**Comment by shakeelmohamed** [comment]
> I've updated package.json file and had to turn off some checks which were interferring with jquery and absolute/relative url consistencies in this project.
@agauniyal can you make a note of those options and create a new issue for fixing them? Or, are they unfixable?

**Comment by shakeelmohamed** [comment]
Looks good. Thanks for doing this work @agauniyal, welcome to the org!

**Comment by agauniyal** [comment]
`absolute url` rule cannot be fixed and I don't believe is an absolute rule to enforce. There's another variable-naming issue rule with jquery which I'll try after a few days (possibly after adding csslint , eslint PR).

**Comment by shakeelmohamed** [comment]
@agauniyal Great, thanks for the effort!

**Pull request merged**